OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectWorld.cs

```diff
@@ -100,12 +124,34 @@ sealed class DreamMetaObjectWorld : IDreamMetaObject {
                     return new DreamValue(_dreamMapManager.Size.Y);
                 case "maxz":
                     return new DreamValue(_dreamMapManager.Levels);
-                //case "address":
-                //    return new(Runtime.Server.Address.ToString());
-                //case "port":
-                //    return new(Runtime.Server.Port);
-                //case "url":
-                //    return new("opendream://" + Runtime.Server.Address + ":" + Runtime.Server.Port);
+                case "address": // By address they mean, the local address we have on the network, not on the internet.
+                    var host = Dns.GetHostEntry(Dns.GetHostName());
+                    var ip_type = AddressFamily.InterNetwork;
+                    if (Display_IPv6)
+                    {
+                        ip_type = AddressFamily.InterNetworkV6;
+                    }
+                    foreach (var ip in host.AddressList)
+                    {
+                        if (ip.AddressFamily == ip_type)
+                        {
+                            return new DreamValue(ip.ToString());
+                        }
+                    }
+                    return DreamValue.Null;
+                case "port":
+                    return new DreamValue(_dreamManager.Port);
+                case "url":
+                    if (_dreamManager.InternetAddress == null)
+                        return DreamValue.Null;
+                    return new("http://" + _dreamManager.InternetAddress + ":" + _dreamManager.Port); // RIP "opendream://"
+                case "internet_address":
```

Comment:
I'm unable to really test this, does it work?

Reply:
I can't actually confirm that it does 😅 

I'm not really set up to do a port-forwarded server at the moment to test it (damned new routers!). Will try to see if I can get that done, though